Tensile/Source/Tools.cpp

```diff
@@ -19,18 +19,40 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
  *******************************************************************************/
+#ifdef _WIN32
+#include <winbase.h>
+#endif
 
-#include "Tools.h"
 #include <cmath>
 #include <ctype.h>
+#include <stdlib.h>
+
+#include "Tools.h"
 
-// QueryPerformanceFrequency is defined to return in units of counts per second
-// QueryPerformanceCounter is defined to return in units of counts
-// counts / (counts / seconds) == seconds, as the native windows time unit
 
-// However, in practice the frequency of QueryPerformanceFrequency() has a
-// resolution measured in nano-seconds on modern hardware, so it possible to
-// return meaningful information on the nano-second time scale
+/*******************************************************************************
+ * Cross platform helpers
+ ******************************************************************************/
+
+const char* read_env(const char* env_var)
+{
+#ifdef _WIN32
+    const DWORD nSize = _MAX_PATH;
+    static thread_local char lpBuffer[nSize];
+    DWORD len = GetEnvironmentVariableA(env_var, lpBuffer, nSize);
+    if (len && len < nSize)
+        return lpBuffer;
+    else 
+        return nullptr; // variable not found or longer than nSize
+#else
+    return std::getenv(env_var);
```

Comment:
Doesn't std::getenv work on windows?

Reply:
Alex did you read the earlier discussion?   It works but there may be issues with replication and modification during tests, hopefully none are applicable to tensile.    If you have wide char paths in the environment variable that would also pose issues as there is a separate API for reading those.

Reply:
Oops, just saw the discussion, that makes sense